### PR TITLE
Add XinaBox CW01 to supported boards

### PR DESCRIPTION
**Pull request by brendanvanbreda**
We have a product based off the ESP8266 module.  
Information of the product can be found on our [store](https://xinabox.cc/products/cw01).
Additional info can be found on our [wiki](https://wiki.xinabox.cc/CW01_-_ESP8266_Wi-Fi_Core).

### CODE REVIEW

**Comment by d-a-v** [comment]
`boards.txt` is now script-generated.
You need to update `tools/boards.txt.py`, run it with option `--allgen` and commit all the re-generated files (like `boards.txt`).

**Comment by brendanvanbreda** [comment]
@d-a-v  i have updated the files as suggest, please note that `--allgen` is not a recognized option.
The correct option is `--boardsgen`.

**Comment by earlephilhower** [comment]
Your generated boards.txt has hundreds of differences unrelated to your board, so something is majorly wrong here.

Can you double-check you had the latest boards.py and that your changes were applied to the latest version of the repo?

**Comment by brendanvanbreda** [comment]
will do

File: tools/boards.txt.py [hunk 1]
```diff
@@ -325,6 +325,23 @@
...
+            '.build.board': 'ESP8266_GENERIC',
+            '.build.variant': 'xinabox',
+            },
+        'macro': [
+            'resetmethod_nodemcu',
+            'flashmode_qio',
+            'flashfreq_40',
+            '4M',
+            ],
+        'desc': [ 'The XinaBox CW01 is an Arduino-compatible Wi-Fi development board powered by Ai-Thinker\'s ESP-12F, clocked at 80 MHz at 3.3V logic. It also has an onboard RGB LED.',
```
Comment (d-a-v) [inline tools/boards.txt.py]: 160MHz is always a possible option for any board.
Maybe you would also add a gpio define in the pins/variant file about your RGB led for your users.

Comment (brendanvanbreda) [reply tools/boards.txt.py]: i will add that info as well

Comment (brendanvanbreda) [reply tools/boards.txt.py]: any ideas as to why the package build keeps failing ?

**Comment by brendanvanbreda** [comment]
Assuming the package build error can be ignored, based on the log from the CI, it would seems everything else is good ?

**Comment by d-a-v** [comment]
You need to commit your `boards.rst` as well as stated in
> Do not create pull-requests for this file only, CI will not accept them.
You *must* edit/modify/run boards.txt.py to regenerate boards.txt.
**All modified files** after running with option "--allgen" must be included in the pull-request.

`git status` will show you what are the modified files.

You also need to update your repository before restarting the generator. I'm afraid you'll get into a git black hole if you merge current master.

Probably the easiest is to re-do the job from scratch.
* Clone the core itself, not your current clone since you modified the `master` branch, to another directory
* add your github fork in remote source list (`git remote add xinabox https://github.com/xinabox/Arduino-ESP8266`)
* make a new branch `git checkout -b newboard`
* copy your `boards.txt.py` from your current repo
* run it with `--allgen`
* check modified files `git status`
* `git add` all of them, then `git commit`
* push that to your github repository in the `newboard` branch `git push xinabox newboard`
* open this branch on github, and press the PR button.

**Comment by brendanvanbreda** [comment]
this can be closed

**Comment by earlephilhower** [comment]
Another clean PR generated for this, closing.

**Pull request closed without merging**